examples/pytorch/rgcn-hetero/entity_classify_mb.py

```diff
@@ -0,0 +1,432 @@
+                    g.srcnodes[srctype].data['x'], ws[etype])
+                funcs[(srctype, etype, dsttype)] = (fn.copy_u('h%d' % i, 'm'), fn.mean('m', 'h'))
+        else:
+            funcs = {}
+            for i, (srctype, etype, dsttype) in enumerate(g.canonical_etypes):
+                if srctype not in xs:
+                    continue
+                g.srcnodes[srctype].data['h%d' % i] = g.srcnodes[srctype].data['x']
+                funcs[(srctype, etype, dsttype)] = (fn.copy_u('h%d' % i, 'm'), fn.mean('m', 'h'))
+        # message passing
+        g.multi_update_all(funcs, 'sum')
+
+        hs = {}
+        for ntype in g.dsttypes:
+            if 'h' in g.dstnodes[ntype].data:
+                hs[ntype] = g.dstnodes[ntype].data['h']
+        def _apply(h):
+            # apply bias and activation
+            if self.self_loop:
+                h = h + th.matmul(xs[ntype], self.loop_weight)
+            if self.bias:
+                h = h + self.h_bias
+            if self.activation:
+                h = self.activation(h)
+            h = self.dropout(h)
+            return h
+        hs = {ntype : _apply(h) for ntype, h in hs.items()}
+        return hs
+
+class RelGraphEmbed(nn.Module):
+    r"""Embedding layer for featureless heterograph."""
+    def __init__(self,
+                 g,
+                 embed_size,
+                 activation=None,
+                 dropout=0.0):
+        super(RelGraphEmbed, self).__init__()
+        self.g = g
+        self.embed_size = embed_size
+        self.activation = activation
+        self.dropout = nn.Dropout(dropout)
+
+        # create weight embeddings for each node for each relation
+        self.embeds = nn.ParameterDict()
+        for ntype in g.ntypes:
+            embed = nn.Parameter(th.Tensor(g.number_of_nodes(ntype), self.embed_size))
+            nn.init.xavier_uniform_(embed, gain=nn.init.calculate_gain('relu'))
+            self.embeds[ntype] = embed
+
+
+    def forward(self, block=None):
+        """Forward computation
+
+        Parameters
+        ----------
+        block : DGLHeteroGraph, optional
+            If not specified, directly return the full graph with embeddings stored in
+            :attr:`embed_name`. Otherwise, extract and store the embeddings to the block
+            graph and return.
+
+        Returns
+        -------
+        DGLHeteroGraph
+            The block graph fed with embeddings.
+        """
+        return self.embeds
+
+class EntityClassify(nn.Module):
+    def __init__(self,
+                 g,
+                 h_dim, out_dim,
+                 num_bases,
+                 num_hidden_layers=1,
+                 dropout=0,
+                 use_self_loop=False):
+        super(EntityClassify, self).__init__()
+        self.g = g
+        self.h_dim = h_dim
+        self.out_dim = out_dim
+        self.rel_names = list(set(g.etypes))
+        self.rel_names.sort()
```

Comment:
It is to keep a deterministic order so that offline evaluation could work. IMO, it can be avoided as long as we use ParameterDict rather than ParameterList. @classicsong what do you think?

Reply:
I see.  I think using `ParameterDict` can make the code more intuitive.